embedded_postgres.go

```diff
@@ -161,10 +176,32 @@ func ensurePortAvailable(port uint32) error {
 	return nil
 }
 
-func userLocationOrDefault(userLocation, cacheLocation string) string {
+func userRuntimePathOrDefault(userLocation, cacheLocation string) string {
 	if userLocation != "" {
 		return userLocation
 	}
 
 	return filepath.Join(filepath.Dir(cacheLocation), "extracted")
 }
+
+func userDataPathOrDefault(userLocation, runtimeLocation string) string {
+	if userLocation != "" {
+		return userLocation
+	}
+
+	return filepath.Join(runtimeLocation, "data")
+}
+
+func dataDirIsValid(dataDir string, version PostgresVersion) bool {
+	pg_version := filepath.Join(dataDir, "PG_VERSION")
+	var v string
+	if d, err := os.ReadFile(pg_version); err != nil {
```

Comment:
go 1.16 feature :disappointed:

Reply:
Yah let's try to support back to 1.13 if possible.